Objects/dictobject.c

```diff
@@ -615,6 +615,52 @@ new_dict_with_shared_keys(PyDictKeysObject *keys)
     return new_dict(keys, values);
 }
 
+
+static PyObject *
+clone_dict(PyDictObject *orig)
```

Comment:
would you mind to rename the function to "clone_combined_dict()"? It requires ma_values==NULL.